src/pages/library/RequestDatasetAccessModal.js

```diff
@@ -12,111 +12,104 @@ import * as Utils from 'src/libs/utils'
+              title,
+              div({ style: { fontSize: '.7rem', marginTop: 5, width: 'fit-content' } }, [
+                _.map(({ email }) => email && h(Link, { key: email, href: `mailto:${email}`, style: { marginTop: 5, display: 'block' } }, [email]), contacts)
+              ])
+            ]),
+            td([
+              Utils.switchCase(access,
+                ['Controlled', () => h(RequestDatasetAccessButton, { title, id, setShowWipModal })],
+                ['Pending', () => span({ style: { fontWeight: 600 } }, ['Request Pending'])],
+                [Utils.DEFAULT, () => span({ style: { fontWeight: 600 } }, ['Permission Granted'])]
+              )
+            ])
+          ]), datasets)
+        )
+      ])
     ])
-  ])
 }
 
-const RequestDatasetAccessButton = ({ title, id }) => {
-  const [requesting, setRequesting] = useState(false)
-  const [requested, setRequested] = useState(false)
-  const [showWipModal, setShowWipModal] = useState(false)
+const RequestDatasetAccessButton = ({ title, id, setShowWipModal }) => {
+  const [status, setStatus] = useState('')
   const signal = Utils.useCancellation()
 
-  return h(Fragment, [
-    showWipModal && h(Modal, {
-      title: `Keep in mind we're still in BETA`,
-      width: '32rem',
-      showButtons: false,
-      onDismiss: () => {}
-    }, [
-      div({ style: { lineHeight: '1.7rem' } }, [
-        `In the near future, when you click on `, strong(['Request Access']),
-        `, the request will be sent to the appropriate approver and the dataset will be marked as `,
-        strong(['Pending']), `.`
-      ]),
-      div({ style: { marginTop: 30, display: 'flex', flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' } }, [
-        `Let's continue testing the BETA Data Catalog!`,
-        h(ButtonPrimary, { onClick: () => { setShowWipModal(false) }, style: { padding: '20px 30px' } }, 'OK')
-      ])
-    ]),
-    h(ButtonPrimary, {
-      disabled: requesting || requested,
-      onClick: () => {
-        const requestAccess = _.flow(
-          Utils.withBusyState(setRequesting),
-          withErrorReporting('Error requesting dataset access')
-        )(async () => {
-          await Ajax(signal).DataRepo.requestAccess(id)
-          setRequested(true)
-        })
-        requestAccess()
-        Ajax().Metrics.captureEvent(`${Events.catalogRequestAccess}:confirmed`, {
-          snapshotId: id,
-          snapshotName: title
-        })
-      }
-    }, [
-      Utils.cond(
-        [requested, () => 'Request Sent'],
-        [requesting, () => 'Sending Request...'],
-        () => 'Request Access'
-      )
-    ])
+  return h(ButtonPrimary, {
+    disabled: status,
+    onClick: withErrorReporting('Error requesting dataset access', async () => {
+      requestAccessEnabled ?
+        await Ajax(signal).DataRepo.requestAccess(id) :
+        setShowWipModal(true)
+      setStatus('Request Sent')
```

Comment:
nit: Now that I am seeing it this way, I am thinking a better name for this may just be `buttonText`, with a default of 'Request Access'